libs/wire-api/src/Wire/API/Routes/Internal/Brig.hs

```diff
@@ -144,24 +145,51 @@ type AccountAPI =
         :> MultiVerb 'POST '[Servant.JSON] RegisterInternalResponses (Either RegisterError SelfProfile)
     )
 
-type MLSAPI = GetClientByKeyPackageRef :<|> GetMLSClients :<|> MapKeyPackageRefs
-
-type GetClientByKeyPackageRef =
-  Summary "Resolve an MLS key package ref to a qualified client ID"
-    :> "mls"
-    :> "key-packages"
-    :> Capture "ref" KeyPackageRef
-    :> MultiVerb
-         'GET
-         '[Servant.JSON]
-         '[ RespondEmpty 404 "Key package ref not found",
-            Respond 200 "Key package ref found" ClientIdentity
-          ]
-         (Maybe ClientIdentity)
+type MLSAPI =
+  "mls"
+    :> ( ( "key-packages" :> Capture "ref" KeyPackageRef
+             :> ( Named
+                    "get-client-by-key-package-ref"
+                    ( Summary "Resolve an MLS key package ref to a qualified client ID"
+                        :> MultiVerb
+                             'GET
+                             '[Servant.JSON]
+                             '[ RespondEmpty 404 "Key package ref not found",
+                                Respond 200 "Key package ref found" ClientIdentity
+                              ]
+                             (Maybe ClientIdentity)
+                    )
+                    :<|> ( "conversation"
+                             :> ( Named
+                                    "put-conversation-by-key-package-ref"
+                                    ( Summary "Associate a conversation with a key package"
+                                        :> ReqBody '[Servant.JSON] (Qualified ConvId)
+                                        :> MultiVerb1
+                                             'PUT
+                                             '[Servant.JSON]
+                                             (Respond 200 "Conversaion associated" ())
+                                    )
+                                    :<|> Named
+                                           "get-conversation-by-key-package-ref"
+                                           ( Summary "Retrieve the conversation associated with a key package"
+                                               :> MultiVerb
+                                                    'GET
+                                                    '[Servant.JSON]
+                                                    [ RespondEmpty 404 "No associated conversaion or bad key package",
```

Comment:
Do we have a maximum line length?  This fits on one line in my normal environment, but it is pretty long, especially for side-by-side diffs.

Reply:
We do not. I tend to wrap them at column 80 or so, but I have no strict or automatic enforcement of this. I find it excessive when lines are over 100 characters long.

Reply:
I'll probably have to break stuff back out in order to get the lines under 100, but I should be able to.